satellite/console/consoleweb/utils.go

```diff
@@ -53,23 +53,25 @@ func getToken(req *http.Request) string {
 }
 
 // getQuery retrieves graphql query from request
-func getQuery(req *http.Request) (query graphqlJSON, err error) {
+func getQuery(w http.ResponseWriter, req *http.Request) (query graphqlJSON, err error) {
 	switch req.Method {
 	case http.MethodGet:
 		query.Query = req.URL.Query().Get(consoleql.Query)
 		return query, nil
 	case http.MethodPost:
-		return queryPOST(req)
+		return queryPOST(w, req)
 	default:
 		return query, errs.New("wrong http request type")
 	}
 }
 
 // queryPOST retrieves graphql query from POST request
-func queryPOST(req *http.Request) (query graphqlJSON, err error) {
+func queryPOST(w http.ResponseWriter, req *http.Request) (query graphqlJSON, err error) {
 	switch typ := req.Header.Get(contentType); typ {
 	case applicationGraphql:
-		body, err := ioutil.ReadAll(req.Body)
+		limitedReader := http.MaxBytesReader(w, req.Body, ContentLengthLimit.Int64())
```

Comment:
This wouldn't limit `json` queries and call close on `limitedReader` rather than `req.Body`.

Reply:
good point. missed that